react/components/contexts/auth-context/auth-context.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from "react";
+import AsyncStorage from "@react-native-community/async-storage";
+import PropTypes from "prop-types";
+import { verify } from "yonius";
+import { API as RipeIdAPI } from "ripe-id-api";
+
+import { RipeContext } from "../ripe-context";
+
+export const AuthContext = React.createContext();
+
+export class AuthProvider extends Component {
+    static contextType = RipeContext;
+
+    static get propTypes() {
+        return {
+            options: PropTypes.shape({
+                clientId: PropTypes.string.isRequired,
+                clientSecret: PropTypes.string.isRequired,
+                redirectUrl: PropTypes.string.isRequired
+            })
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.ripeIdApi = new RipeIdAPI(this.props.options);
+        this.state = {
+            loaded: false,
+            logoutMessage: null,

```

Comment:
Why is this needed?

Reply:
I replicated from the `OAuthContext`. It is used here: https://github.com/ripe-tech/ripe-id-mobile/blob/master/app.js#L139